L1Trigger/L1THGCal/interface/hw_interfaces/HGCalHardwareInterface.h

```diff
@@ -0,0 +1,59 @@
+#ifndef __L1Trigger_L1THGCal_HGCalHardwareInterface_h__
+#define __L1Trigger_L1THGCal_HGCalHardwareInterface_h__
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/L1THGCal/interface/HGCalCluster.h"
+#include "DataFormats/L1THGCal/interface/HGCalMulticluster.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+class HGCalHardwareInterface{
+  
+ public:
+  /* mapping is retrieved from: https://twiki.cern.ch/twiki/bin/viewauth/CMS/HGCALTriggerArchitectureDesignAndImplementation */
+  struct ClusterDataFormat{
+    enum CDF {ET = 0, CenterY = 8, CenterX = 20, Flags = 32, SizeY = 38, SizeX = 46, LocalMax = 54, NoTriggerCells = 56, LocalMax1RelX = 64,  LocalMax0ET = 72, LocalMax0RelY = 80, LocalMax0RelX = 88, LocalMax2RelY = 96,  LocalMax2RelX = 104,  LocalMax1ET = 112,  LocalMax1RelY = 120,  LocalMax3ET = 128, LocalMax3RelY = 136,  LocalMax3RelX = 144,  LocalMax2ET = 152 };
+  };
+  struct MulticlusterDataFormat{
```

Comment:
In my opinion Clusters and MultiClusters are purely software objects (the first contains a vector of trigger cells, while the second contains a vector of clusters), and we would have one single class if there was not the need to track the constituents in the software. This is not linked to an hardware architecture.  
In the future we could have Clusters object with the format currently called `MultiClusterDataFormat`.

Also, the data format used here is really linked to 2D objects for the first one (x,y-related values, but no z-related values, and no projective variables) and for 3D objects for the second one (projective positions, longitudinal information)